### PR TITLE
Use jed1x json format to correctly handle pluralization

### LINKED ISSUE #702: [i18n] Pluralized translations are not downloaded
## Description

Plural translations for languages that use more than one noun variant for plurals is incorrect.

In Russian, there are 3 variants of 'images' plural, depending on the number. 

The 3 variants are available [on `translate.wordpress.org` ](https://translate.wordpress.org/projects/meta/openverse/ru/default/?filters%5Bstatus%5D=either&filters%5Boriginal_id%5D=12384747&filters%5Btranslation_id%5D=88548343):

Screen Shot 2022-01-25 at 7 47 17 AM
However, the `ngx` `json` file exported from GlotPress, only contains a single variant, so the translation is often incorrect:

Screen Shot 2022-01-25 at 7 46 51 AM

## Reproduction

1. Look at the `ru.json` file inside `src/locales/` directory.
2. Find  the following key that uses plurals: `"browse-page.all-result-count-more"`.
3. See error: only 1 variant  "Более {localeCount} изображения", instead of the following variants available in the `po` export:
```
#. Do not translate words between ### ###.
msgctxt "browse-page.image-result-count-more"
msgid "Over ###localeCount### image results"
msgid_plural "Over ###localeCount### image results"
msgstr[0] "Более ###localeCount### изображения"
msgstr[1] "Более ###localeCount### изображений"
msgstr[2] "Более ###localeCount### изображений"
```

## Expectation

All of the results should be downloaded. So, for `browse-page.image-result-count-more` key, the value should be:
`"Более {localeCount} изображения|Более {localeCount} изображений|Более {localeCount} изображений"`

## Additional context

Ideally, the `json` file should have all the available variants. However, it might be faster for us to use `.po` files, and convert them back to the `json` format.

## Resolution

- [ ] 🙋 I would be interested in resolving this bug.

### PR DESCRIPTION
**Pull request by obulat**
## Fixes

Fixes #702 by @obulat

## Description

This PR uses  a more modern JSON format for downloading the translations from `translate.wordpress.org`, `jed1x`. This enables pluralization (the NGX format did not have any information about pluralization).

## Testing Instructions

Run `pnpm run i18n:get-translations` to update the translations for all locales. You should see that the keys that need to be pluralized (eg. `waveform.current-time`, `browse-page.all-result-count-more`), have several values joined with a pipe if translations for that locale has been done.

## Checklist

- [ ] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [ ] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [ ] My commit messages follow [best practices][best_practices].
- [ ] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: src/locales/en.json [hunk 1]
```diff
@@ -405,7 +405,7 @@
   "browse-page": {
     "all-no-results": "No results",
     "all-result-count": "{localeCount} result|{localeCount} results",
-    "all-result-count-more": "Over {localeCount} results",
+    "all-result-count-more": "Over {localeCount} result|Over {localeCount} results",
```
Comment (sarayourfriend) [inline src/locales/en.json:408]: I wonder if there's some kind of ESLint rule we could write to emit a warning or something when we find a translation string that should probably have plural variations... like maybe something basic, if the English string ends with `s` and also contains a variable interpolation? 🤔

File: src/locales/scripts/jed1x-json-to-json.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+// https://github.com/messageformat/Jed/blob/351c47d5c57c5c81e418414c53ca84075c518edb/jed.js#L117
+const SPLIT_CHAR = String.fromCharCode(4)
+
+function jed1xJsonToJson(jed1xObject) {
+  const result = {}
+  Object.entries(jed1xObject?.locale_data?.messages).forEach(([key, value]) => {
+    const cleanedKey = key.slice(0, key.indexOf(SPLIT_CHAR))
+    if (value.length > 0) {
+      const cleanedValue = value.length === 1 ? value[0] : value.join('|')
+      return set(result, cleanedKey, cleanedValue)
```
Comment (sarayourfriend) [inline src/locales/scripts/jed1x-json-to-json.js:40]: Is `lodash.set` necessary here? Could we use the native `Object.assign` instead?

```suggestion
      return Object.assign(result, { [cleanedKey]:  cleanedValue })
```

**Review by sarayourfriend — APPROVED** [review]
LGTM! Good catch 🙂

File: src/locales/scripts/jed1x-json-to-json.js (hunk 2 above)
Comment (obulat) [reply src/locales/scripts/jed1x-json-to-json.js:40]: To be honest, I just adapted the code  that Zack wrote for NGX format. I think it converts the nested keys that have several dots in them into a nested object, but I'm not sure 🤔

File: src/locales/en.json (hunk 1 above)
Comment (obulat) [reply src/locales/en.json:408]: That's an interesting idea, @sarayourfriend !

File: src/locales/scripts/jed1x-json-to-json.js (hunk 2 above)
Comment (sarayourfriend) [reply src/locales/scripts/jed1x-json-to-json.js:40]: Ohhhh that makes sense, the `cleanedKey` is a path 😭 Never mind then!

File: src/locales/scripts/jed1x-json-to-json.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+ *  "browse-page.search-form.button\u0004Search": [],
+ * }
+ * To:
+ * {
+ *  "browse-page: {
+ *   "load": "Загрузить ещё результаты"
+ *   "all-result-count-more": "Более ###localeCount### результата|Более ###localeCount### результатов|Более ###localeCount### результатов",
+ * }
+ *
+ */
```
Comment (zackkrida) [inline src/locales/scripts/jed1x-json-to-json.js:28]: This is _excellent_ documentation, great!

**Review by zackkrida — APPROVED** [review]
The documentation made it very easy to understand the changes here. LGTM!

**Pull request merged**